pkg/tests/apis/alerting/notifications/receivers/receiver_test.go

```diff
@@ -0,0 +1,927 @@
+			route := legacyCli.GetRoute(t)
+			route.Routes[0].Receiver = ""
+			legacyCli.UpdateRoute(t, route, true)
+
+			err = adminClient.Delete(ctx, receiver.Name, v1.DeleteOptions{})
+			require.Truef(t, errors.IsConflict(err), "Expected Conflict, got: %s", err)
+		})
+	})
+}
+
+func TestIntegrationCRUD(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ctx := context.Background()
+	helper := getTestHelper(t)
+
+	adminK8sClient, err := versioned.NewForConfig(helper.Org1.Admin.NewRestConfig())
+	require.NoError(t, err)
+	adminClient := adminK8sClient.NotificationsV0alpha1().Receivers("default")
+
+	var defaultReceiver *v0alpha1.Receiver
+	t.Run("should list the default receiver", func(t *testing.T) {
+		items, err := adminClient.List(ctx, v1.ListOptions{})
+		require.NoError(t, err)
+		assert.Len(t, items.Items, 1)
+		defaultReceiver = &items.Items[0]
+		assert.Equal(t, "grafana-default-email", defaultReceiver.Spec.Title)
+		assert.NotEmpty(t, defaultReceiver.UID)
+		assert.NotEmpty(t, defaultReceiver.Name)
+		assert.NotEmpty(t, defaultReceiver.ResourceVersion)
+
+		defaultReceiver, err = adminClient.Get(ctx, defaultReceiver.Name, v1.GetOptions{})
+		require.NoError(t, err)
+		assert.NotEmpty(t, defaultReceiver.UID)
+		assert.NotEmpty(t, defaultReceiver.Name)
+		assert.NotEmpty(t, defaultReceiver.ResourceVersion)
+		assert.Len(t, defaultReceiver.Spec.Integrations, 1)
+	})
+
+	t.Run("should be able to update default receiver", func(t *testing.T) {
+		require.NotNil(t, defaultReceiver)
+		newDefault := defaultReceiver.DeepCopy()
+		newDefault.Spec.Integrations = append(newDefault.Spec.Integrations, createIntegration(t, "line"))
+
+		updatedReceiver, err := adminClient.Update(ctx, newDefault, v1.UpdateOptions{})
+		require.NoError(t, err)
+
+		expected := newDefault.DeepCopy()
+		expected.Spec.Integrations[0].Uid = updatedReceiver.Spec.Integrations[0].Uid // default integration does not have UID before first update
+		lineIntegration := expected.Spec.Integrations[1]
+		lineIntegration.SecureFields = map[string]bool{
+			"token": true,
+		}
+		lineIntegration.Settings.Remove("token")
+		assert.Equal(t, "LINE", updatedReceiver.Spec.Integrations[1].Type) // this type is in the schema but not in backend
+		lineIntegration.Type = "LINE"
+		lineIntegration.Uid = updatedReceiver.Spec.Integrations[1].Uid
+		expected.Spec.Integrations[1] = lineIntegration
+
+		assert.Equal(t, expected.Spec, updatedReceiver.Spec)
+	})
+
+	t.Run("should fail to create receiver with the existing name", func(t *testing.T) {
+		newReceiver := &v0alpha1.Receiver{
+			ObjectMeta: v1.ObjectMeta{
+				Namespace: "default",
+			},
+			Spec: v0alpha1.ReceiverSpec{
+				Title:        defaultReceiver.Spec.Title,
+				Integrations: nil,
+			},
+		}
+		_, err := adminClient.Create(ctx, newReceiver, v1.CreateOptions{})
+		require.Truef(t, errors.IsBadRequest(err), "Expected BadRequest, got: %s", err)
```

Comment:
```suggestion
		require.Truef(t, errors.IsConflict(err), "Expected Conflict, got: %s", err)
```

This should actually be a conflict, there's a mixup in our errors: https://github.com/grafana/grafana/blob/32f06c6d9ce3d1768eb5ec36e2695a7f86e4c461/pkg/services/ngalert/notifier/legacy_storage/errors.go#L10-L11

`ErrReceiverExists` should be `Conflict` and `ErrReceiverInvalid` should be `BadRequest`, not the other way around.

This also means there's a gap in the integration tests making sure we fail when trying to create invalid receivers. I'll add a snippet below that covers this.